### ISSUE TITLE
Wrong overload resolution when using user-defined types

### ISSUE THREAD

**Issue by georgiy-belyanin** [EmmyLuaLs/emmylua-analyzer-rust#360]
Wrong diagnostics are issued if we use user-defined types combined with overloads when no such arguments are supplied.

```lua
---@alias buz number

---@param a buz
---@overload fun(): number
function test(a)
end

local c = test({'test'}) -- Wrong diagnostic: expected 0 parameters found 1

---@return number
---@overload fun(a: buz)
function test2(a)
end

local c = test2({'test'}) -- Ok
```
